src/plugins/kibana_utils/public/render_complete/render_complete_dispatcher.ts

```diff
@@ -64,7 +64,6 @@ export class RenderCompleteDispatcher {
     if (!this.el) return;
     this.count++;
     this.el.setAttribute('data-render-complete', 'true');
-    this.el.setAttribute('data-loading', 'false');
```

Comment:
The `waitForRender` method expects no `data-loading` attribute **at all** for an element to be considered rendered for functional tests - that is, `data-loading=false` is still considered to be a loading element. Therefore, we should not be setting `data-loading` to `false` here, because otherwise panels with errors are never considered to be done rendering: 

https://github.com/elastic/kibana/blob/28efce79557a89f81139c2e71f4c68416201bcf4/test/functional/services/renderable.ts#L13

https://github.com/elastic/kibana/blob/28efce79557a89f81139c2e71f4c68416201bcf4/test/functional/services/renderable.ts#L44-L47

And, even once they recover from an error, the `data-loading=false` parameter sticks around because it is never removed. All `data-loading` state should instead by handled by the `render` method of embeddable panels, instead.

Reply:
Thanks for the explanation, this really makes sense!